drivers/retained_mem/retained_mem_nrf_gpregret.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_gpregret
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/retained_mem.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(retained_mem_nrf_gpregret, CONFIG_RETAINED_MEM_LOG_LEVEL);
+
+#ifdef CONFIG_MULTITHREADING
+struct nrf_gpregret_data {
+	struct k_mutex lock;
+};
+#endif
+
+struct nrf_gpregret_config {
+	uint32_t addr;
```

Comment:
Nonblocking: wouldn't it be simpler to declare this as a `uint8_t*`?

Reply:
Fixed